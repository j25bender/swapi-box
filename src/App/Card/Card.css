.card {
  display: flex;
  flex-flow: column wrap;
  height: 200px;
  width: 150px;
  border: 1px solid lightslategrey;
  text-align: center;
  margin: 10px;
  font-size: 16px;
  background-color: rgba(0, 0, 0, 0.1); }

.card:hover {
  border: 1px solid #3137e0;
  box-shadow: 0 0 2px 2px #3137e0; }

.card h6 {
  color: lightslategrey;
  margin: auto;
  overflow: hidden; }

.card span {
  color: ghostwhite; }

img {
  max-height: 10px;
  max-width: 10px;
  padding-right: 5px; }
