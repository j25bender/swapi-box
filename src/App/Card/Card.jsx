import React from 'react';
import PropTypes from "prop-types";

const Card = () => {
    return (
        <div className="card">
            {
                this.props.people &&
                    <div className="people">
                        <div className="name">{ this.props.name }</div>
                        <div className="homeworldName">{ this.props.homeworldName }</div>
                        <div className="homeworldPop">{ this.props.homeworldPop }</div>
                        <div className="speciesType">{ this.props.speciesType }</div>
                    </div>
            }
                
            {
                this.props.planets &&
                    <div className="planet">
                        <div className="name">{ this.props.name }</div>
                        <div className="terrain">{ this.props.terrain }</div>
                        <div className="population">{ this.props.population }</div>
                        <div className="climate">{ this.props.climate }</div>
                        <div className="residents">{ this.props.residents }</div>
                    </div>
            }

            {
                this.props.vehicles &&
                    <div className="vehicle">
                        <div className="name">{ this.props.name }</div>
                        <div className="model">{ this.props.model }</div>
                        <div className="passengers">{ this.props.vehicleClass }</div>
                    </div>
            }
        </div>
    )
}

export default Card