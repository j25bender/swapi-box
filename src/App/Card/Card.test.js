import React from 'react'
import Enzyme, {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Card from './Card'

Enzyme.configure({ adapter: new Adapter() })

describe('Card', () => {
    const  mockPlanet = { info: [
        "Climate": "arid", 
        "Name": "Tatooine", 
        "Population": "200000", 
        "Residents": ["Luke Skywalker"], 
        "Terrain": "desert"
    ]}

    const mockselectCard = jest.fn()

    it('matches the snapshot', () => {
        const wrapper = shallow(<Card data={mockPlanet} selectCard={mockselectCard} />)
        
        expect(wrapper).toMatchSnapshot()
    })
    
})