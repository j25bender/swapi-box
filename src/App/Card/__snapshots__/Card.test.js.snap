// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "info": Array [
              "Climate",
              "Name",
              "Population",
              "Residents",
              "Terrain",
            ],
          }
    }
    selectCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="card"
        onClick={[Function]}
>
        <h6>
                0
                :  
                <span>
                        Climate
                </span>
        </h6>
        <h6>
                1
                :  
                <span>
                        Name
                </span>
        </h6>
        <h6>
                2
                :  
                <span>
                        Population
                </span>
        </h6>
        <h6>
                3
                :  
                <span>
                        Residents
                </span>
        </h6>
        <h6>
                4
                :  
                <span>
                        Terrain
                </span>
        </h6>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h6>
            0
            :  
            <span>
                        Climate
            </span>
</h6>,
          <h6>
            1
            :  
            <span>
                        Name
            </span>
</h6>,
          <h6>
            2
            :  
            <span>
                        Population
            </span>
</h6>,
          <h6>
            3
            :  
            <span>
                        Residents
            </span>
</h6>,
          <h6>
            4
            :  
            <span>
                        Terrain
            </span>
</h6>,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "0",
              ":  ",
              <span>
                Climate
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "0",
            ":  ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Climate",
              },
              "ref": null,
              "rendered": "Climate",
              "type": "span",
            },
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "1",
              ":  ",
              <span>
                Name
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "1",
            ":  ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name",
              },
              "ref": null,
              "rendered": "Name",
              "type": "span",
            },
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2",
              ":  ",
              <span>
                Population
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "2",
            ":  ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Population",
              },
              "ref": null,
              "rendered": "Population",
              "type": "span",
            },
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "3",
              ":  ",
              <span>
                Residents
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "3",
            ":  ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Residents",
              },
              "ref": null,
              "rendered": "Residents",
              "type": "span",
            },
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "4",
              ":  ",
              <span>
                Terrain
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "4",
            ":  ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Terrain",
              },
              "ref": null,
              "rendered": "Terrain",
              "type": "span",
            },
          ],
          "type": "h6",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="card"
          onClick={[Function]}
>
          <h6>
                    0
                    :  
                    <span>
                              Climate
                    </span>
          </h6>
          <h6>
                    1
                    :  
                    <span>
                              Name
                    </span>
          </h6>
          <h6>
                    2
                    :  
                    <span>
                              Population
                    </span>
          </h6>
          <h6>
                    3
                    :  
                    <span>
                              Residents
                    </span>
          </h6>
          <h6>
                    4
                    :  
                    <span>
                              Terrain
                    </span>
          </h6>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h6>
              0
              :  
              <span>
                            Climate
              </span>
</h6>,
            <h6>
              1
              :  
              <span>
                            Name
              </span>
</h6>,
            <h6>
              2
              :  
              <span>
                            Population
              </span>
</h6>,
            <h6>
              3
              :  
              <span>
                            Residents
              </span>
</h6>,
            <h6>
              4
              :  
              <span>
                            Terrain
              </span>
</h6>,
          ],
          "className": "card",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "0",
                ":  ",
                <span>
                  Climate
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "0",
              ":  ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Climate",
                },
                "ref": null,
                "rendered": "Climate",
                "type": "span",
              },
            ],
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "1",
                ":  ",
                <span>
                  Name
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "1",
              ":  ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Name",
                },
                "ref": null,
                "rendered": "Name",
                "type": "span",
              },
            ],
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2",
                ":  ",
                <span>
                  Population
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "2",
              ":  ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Population",
                },
                "ref": null,
                "rendered": "Population",
                "type": "span",
              },
            ],
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "3",
                ":  ",
                <span>
                  Residents
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "3",
              ":  ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Residents",
                },
                "ref": null,
                "rendered": "Residents",
                "type": "span",
              },
            ],
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "4",
                ":  ",
                <span>
                  Terrain
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "4",
              ":  ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Terrain",
                },
                "ref": null,
                "rendered": "Terrain",
                "type": "span",
              },
            ],
            "type": "h6",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
